server/helpers/adapters_tracker.go

```diff
@@ -119,11 +124,90 @@ func (a *AdaptersTracker) DeployAdapter(ctx context.Context, adapter models.Adap
 			return ErrAdapterAdministration(err)
 		}
 
+	case "kubernetes":
+		kubeClient, err := meshkitkube.New([]byte(""))
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		port, _ := strconv.Atoi(adapter.Location)
+
+		// Create a deployment
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      adapter.Name,
+				Namespace: core.MesheryNamespace,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": adapter.Name,
+					},
+				},
+				Replicas: int32Ptr(1),
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"app": adapter.Name,
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  adapter.Name,
+								Image: "layer5/" + adapter.Name + ":stable-latest",
+								Ports: []corev1.ContainerPort{
+									{
+										ContainerPort: int32(port),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		_, err = kubeClient.KubeClient.AppsV1().Deployments(core.MesheryNamespace).Create(context.Background(), deployment, metav1.CreateOptions{})
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      adapter.Name,
+				Namespace: core.MesheryNamespace,
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: map[string]string{
+					"app": adapter.Name,
+				},
+				Ports: []corev1.ServicePort{
+					{
+						Name:     "http",
+						Protocol: corev1.ProtocolTCP,
+						Port:     int32(port),
+						TargetPort: intstr.IntOrString{
+							Type:   intstr.Int,
+							IntVal: int32(port),
+						},
+					},
+				},
+				Type: corev1.ServiceTypeClusterIP, // or use appropriate service type
+			},
+		}
+
+		_, err = kubeClient.KubeClient.CoreV1().Services(core.MesheryNamespace).Create(context.Background(), service, metav1.CreateOptions{})
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
 	// switch to default case if the platform specified is not supported
 	default:
 		return ErrAdapterAdministration(fmt.Errorf("the platform %s is not supported currently. The supported platforms are:\ndocker\nkubernetes", platform))
 	}
 
+	adapter.Location = "localhost:" + adapter.Location
```

Comment:
We can’t assume “localhost”.

Reply:
Because it won't work in a multi-node cluster where meshery server and deployed adapter are different node?